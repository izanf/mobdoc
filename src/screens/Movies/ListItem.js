import React from 'react';
import { ListItem as Wrapper } from 'react-native-material-ui';
import { withNavigation } from 'react-navigation';

const ListItem = ({ data: { item: movie }, navigation }) => {
  return (
    <Wrapper
      key={movie['episode_id']}
      divider
      centerElement={{
        primaryText: movie.title,
      }}
      onPress={() => navigation.navigate('MovieDetails', { movie })}
    />
  );
};

export default withNavigation(ListItem);
