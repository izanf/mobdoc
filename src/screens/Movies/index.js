import React, { Component } from 'react';
import { View, FlatList } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Toolbar } from 'react-native-material-ui';

import { fetchMovies } from './../../reducers/movies';

import ListItem from './ListItem';

class Movies extends Component {
  componentDidMount () {
    const { fetchMovies } = this.props;

    fetchMovies();
  }

  render () {
    const { movies } = this.props;

    return (
      <View>
        <Toolbar
          centerElement="Filmes Starwars"
          searchable={{
            autoFocus: true,
            placeholder: 'Search',
          }}
          style={{ backgroundColor: '#1D3062' }}
        />
        <FlatList
          data={movies}
          renderItem={item => <ListItem data={item} />}
          keyExtractor={(_, index) => index.toString()}
        />
      </View>
    );
  }
}

export default connect(state => ({
  loading: state.movies.loading,
  movies: state.movies.data
}),
dispatch => bindActionCreators({
  fetchMovies
}, dispatch))(Movies);
