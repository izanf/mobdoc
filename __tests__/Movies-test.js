import React from 'react';
import Movies from './../src/screens/Movies';

import renderer from 'react-test-renderer';

test('Movies renders correctly', () => {
  const tree = renderer.create(<Movies />).toJSON();
  expect(tree).toMatchSnapshot();
});