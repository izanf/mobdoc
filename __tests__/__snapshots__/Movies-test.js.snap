// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies renders correctly 1`] = `
<View>
  <Text>
    Movies
  </Text>
</View>
`;
